src/database/firebase_object_factory.ts

```diff
@@ -17,13 +18,13 @@ export function FirebaseObjectFactory (
     isRef: () => ref = <firebase.database.Reference>absoluteUrlOrDbRef
   });
 
-  return new FirebaseObjectObservable((obs: Observer<any>) => {
+  return observeOn.call(new FirebaseObjectObservable((obs: Observer<any>) => {
```

Comment:
I think this should match list_factory stylistically, and assign the observable to a const before calling `observeOn`. Makes it more readable.